src/app/games/epaminondas/epaminondas.component.ts

```diff
@@ -313,40 +316,42 @@ export class EpaminondasComponent extends RectangularGameComponent<EpaminondasRu
             return this.tryMove(move);
         }
     }
-    private async moveFirstPiece(PLAYER: Player): Promise<MGPValidation> {
-        this.firstPiece = this.firstPiece.getNext(this.phalanxDirection, 1);
+    private async moveFirstPiece(player: Player): Promise<MGPValidation> {
+        this.firstPiece = this.firstPiece.getNext(this.phalanxDirection.get(), 1);
         if (this.firstPiece.equals(this.lastPiece)) {
-            this.lastPiece = new Coord(-15, -1);
-            this.validExtensions = this.getFirstPieceExtensions(PLAYER);
-            this.phalanxDirection = null;
-            this.phalanxMiddles = [];
+            this.moveOnlyPiece(player);
         } else {
             this.phalanxMiddles = this.phalanxMiddles.slice(1);
-            this.validExtensions = this.getPhalanxValidExtensions(PLAYER);
+            this.validExtensions = this.getPhalanxValidExtensions(player);
         }
 
         this.phalanxValidLandings = this.getPhalanxValidLandings();
         return MGPValidation.SUCCESS;
     }
-    private async moveLastPiece(PLAYER: Player): Promise<MGPValidation> {
-        this.lastPiece = this.lastPiece.getPrevious(this.phalanxDirection, 1);
+    private async moveLastPiece(player: Player): Promise<MGPValidation> {
+        this.lastPiece = this.lastPiece.getPrevious(this.phalanxDirection.get(), 1);
         if (this.firstPiece.equals(this.lastPiece)) {
-            this.lastPiece = new Coord(-15, -1);
-            this.validExtensions = this.getFirstPieceExtensions(PLAYER);
-            this.phalanxDirection = null;
-            this.phalanxMiddles = [];
+            return this.moveOnlyPiece(player);
         } else {
             this.phalanxMiddles = this.firstPiece.getCoordsToward(this.lastPiece);
-            this.validExtensions = this.getPhalanxValidExtensions(PLAYER);
+            this.validExtensions = this.getPhalanxValidExtensions(player);
         }
 
         this.phalanxValidLandings = this.getPhalanxValidLandings();
         return MGPValidation.SUCCESS;
     }
+    private async moveOnlyPiece(player: Player): Promise<MGPValidation> {
+        this.lastPiece = new Coord(-15, -1);
```

Comment:
make it optional rather that fake values :/ ?

Reply:
euh oui très clairement :D fait !